Source/Core/CameraLimiter.js

```diff
@@ -0,0 +1,301 @@
+    'use strict';
+
+    /**
+     * The CameraLimiter is used to limit a {@Camera}'s position (using a bounding object) and orientation (defined by {@HeadingPitchRange}.
+     *
+     * @alias CameraLimiter
+     * @constructor
+     *
+     * @param {AxisAlignedBoundingObject|BoundingRectangle|BoundingSphere|OrientedBoundingBox} [options.boundingObject = undefined] Assigned to the boundingObject attribute which can be used to specify limits on locations.
+     * @param {HeadingPitchRoll} [options.minimumHeadingPitchRoll=undefined] Assigned to the minHeadingPitchRoll attribute which can be used to limit the minimum orientation.
+     * @param {HeadingPitchRoll} [options.maximumHeadingPitchRoll=undefined] Assigned to the maxHeadingPitchRoll attribute which can be used to limit the maximum orientation.
+     */
+    function CameraLimiter(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        this.boundingObject = options.boundingObject;
+
+        this.minHeadingPitchRoll = HeadingPitchRoll.clone(options.minimumHeadingPitchRoll);
+        this.maxHeadingPitchRoll = HeadingPitchRoll.clone(options.maximumHeadingPitchRoll);
+    }
+
+    /**
+     * If inputted position is valid in bounding object, returns position. Otherwise returns closest valid location on the bounding object to the position.
+     *
+     * @param {Cartesian3|Cartographic} position The position we are using to check for a valid location in the bounding object.
+     * @return {Cartesian3|Cartographic} A valid location.
+     */
+    CameraLimiter.prototype.closestLocationTo = function(position) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('positionToCheck', position);
+        if (!(position instanceof Cartesian3) && !(position instanceof Cartographic)) {
+            throw new DeveloperError('positionToCheck required to be of type Cartesian3 or Cartographic.');
+        }
+        //>>includeEnd('debug');
+
+        if (!defined(this.boundingObject)) {
+            return position;
+        }
+
+        var cartesianPosition = (position instanceof Cartographic) ? Cartographic.toCartesian(position) : position;
+
+        if (this.boundingObject instanceof AxisAlignedBoundingBox) {
+            cartesianPosition = this._closestLocationInAxisAligned(cartesianPosition);
+        } else if (this.boundingObject instanceof BoundingRectangle) {
+            cartesianPosition = this._closestLocationInBoundingRectangle(cartesianPosition);
+        } else if (this.boundingObject instanceof BoundingSphere) {
+            cartesianPosition = this._closestLocationInBoundingSphere(cartesianPosition);
+        } else if (this.boundingObject instanceof OrientedBoundingBox) {
+            cartesianPosition = this._closestLocationInOrientedBoundingBox(cartesianPosition);
+        } else {
+            //>>includeStart('debug', pragmas.debug);
+            throw new DeveloperError('Bounding Object not of allowed type. '
```

Comment:
Maybe consider moving the `getClosestBoundedPosition` to each of these classes, and just calling `this.boundingObject.getClosestBoundedPosition() instead of all these cases.